### PR TITLE
Adds file_name setter for setting the default file name (for windows)

### PR DESCRIPTION
**Pull request by AngelOfSol**
As title.

Let me know if you think there's anything that needs to be adjusted!

### CODE REVIEW

File: src/backend/win_cid/file_dialog/dialog_ffi.rs [hunk 1]
```diff
@@ -88,11 +86,12 @@ impl IDialog {
...
-
-                for e in f.extensions.iter() {
-                    ext_string += &format!("*.{};", e);
-                }
+                let ext_string = f
+                    .extensions
+                    .iter()
+                    .map(|item| format!("*.{}", item))
+                    .collect::<Vec<_>>()
+                    .join(";");
```
Comment (PolyMeilex) [inline src/backend/win_cid/file_dialog/dialog_ffi.rs:94]: So... the last item should not have `;` at the end. Did I understand that right?

**Comment by PolyMeilex** [comment]
Thanks for the PR!
I will implement Linux support for this right after this is merged.
Not sure if this is possible on macOS tho, I will have to investigate it later.

Just let me know it I understood the `;` thing right, and I will merge.

File: src/backend/win_cid/file_dialog/dialog_ffi.rs (hunk 1 above)
Comment (AngelOfSol) [reply src/backend/win_cid/file_dialog/dialog_ffi.rs:94]: Ahh, yeah.  I was gonna make a separate PR fixing this!  VSCode for example only put a `;` if there are multiple items in the list.

Comment (AngelOfSol) [reply src/backend/win_cid/file_dialog/dialog_ffi.rs:94]: I forgot to make a separate branch for this fix, so it got picked up here.

Comment (PolyMeilex) [reply src/backend/win_cid/file_dialog/dialog_ffi.rs:94]: No problem, commit name is descriptive enough. I just wanted to make sure that I understand this fix correctly.

**Pull request merged**